test/plots/barPlotTests.ts

```diff
@@ -839,138 +892,192 @@ describe("Plots", () => {
+        };
+      };
 
       beforeEach(() => {
         svg = TestMethods.generateSVG();
-        data = [{x: "foo", y: 5}, {x: "bar", y: 640}, {x: "zoo", y: 12345}];
-        xScale = new Plottable.Scales.Category();
         yScale = new Plottable.Scales.Linear();
+        yScale.domain(DEFAULT_DOMAIN);
+        xScale = new Plottable.Scales.Linear();
+        xScale.domain(DEFAULT_DOMAIN);
+
+        let data = [
+          { x: -4, y: -4 },
+          { x: -2, y: -0.1},
+          { x: 0, y: 0 },
+          { x: 2, y: 0.1 },
+          { x: 4, y: 4 }
+        ];
+
+        barPlot = new Plottable.Plots.Bar<number, number>(Plottable.Plots.Bar.ORIENTATION_VERTICAL);
         dataset = new Plottable.Dataset(data);
-        plot = new Plottable.Plots.Bar<string, number>();
-        plot.addDataset(dataset);
-        plot.x((d) => d.x, xScale);
-        plot.y((d) => d.y, yScale);
+        barPlot.addDataset(dataset);
+        barPlot.x((d) => d.x, xScale);
+        barPlot.y((d) => d.y, yScale);
+        barPlot.renderTo(svg);
       });
 
       it("bar labels disabled by default", () => {
-        plot.renderTo(svg);
-        let texts = svg.selectAll("text")[0].map((n: any) => d3.select(n).text());
-        assert.lengthOf(texts, 0, "by default, no texts are drawn");
+        barPlot.renderTo(svg);
+        let texts = barPlot.content().selectAll("text");
+        assert.strictEqual(texts.size(), 0, "by default, no texts are drawn");
         svg.remove();
       });
 
       it("bar labels render properly", () => {
-        plot.renderTo(svg);
-        plot.labelsEnabled(true);
-        let texts = svg.selectAll("text")[0].map((n: any) => d3.select(n).text());
-        assert.lengthOf(texts, 2, "both texts drawn");
-        assert.strictEqual(texts[0], "640", "first label is 640");
-        assert.strictEqual(texts[1], "12345", "first label is 12345");
+        barPlot.renderTo(svg);
+        barPlot.labelsEnabled(true);
+        let texts = barPlot.content().selectAll("text");
+        let data = dataset.data();
+        assert.strictEqual(texts.size(), data.length, "one label drawn per datum");
+        texts.each(function(d, i) {
+          assert.strictEqual(d3.select(this).text(), data[i].y.toString(), `by default, label text is the bar's value (index ${i})`);
+        });
         svg.remove();
       });
 
       it("bar labels hide if bars too skinny", () => {
-        plot.labelsEnabled(true);
-        plot.renderTo(svg);
-        plot.labelFormatter((n: number) => n.toString() + (n === 12345 ? "looong" : ""));
-        let texts = svg.selectAll("text")[0].map((n: any) => d3.select(n).text());
-        assert.lengthOf(texts, 0, "no text drawn");
+        svg.attr("width", 100);
+        barPlot.labelsEnabled(true);
+        barPlot.renderTo(svg);
+        let texts = barPlot.content().selectAll("text");
+        assert.strictEqual(texts.size(), 0, "no labels drawn");
```

Comment:
I think `size()` is better, since it would provide more information about how many extra were drawn.